go/heif/metadata_test.go

```diff
@@ -0,0 +1,113 @@
+package heif
+
+import (
+	"fmt"
+	"path"
+	"testing"
+)
+
+func TestExifCount(t *testing.T) {
+	ctx, err := NewContext()
+	if err != nil {
+		t.Fatalf("Can't create context: %s", err)
+	}
+
+	filename := path.Join("..", "..", "examples", "mont.heic")
+	if err := ctx.ReadFromFile(filename); err != nil {
+		t.Fatalf("Can't read from %s: %s", filename, err)
+	}
+
+	if count := ctx.GetNumberOfTopLevelImages(); count != 16 {
+		t.Errorf("Expected %d top level images, got %d", 16, count)
+	}
+	if ids := ctx.GetListOfTopLevelImageIDs(); len(ids) != 16 {
+		t.Errorf("Expected %d top level image ids, got %+v", 16, ids)
+	}
+	if _, err := ctx.GetPrimaryImageID(); err != nil {
+		t.Errorf("Expected a primary image, got %s", err)
+	}
+	handle, err := ctx.GetPrimaryImageHandle()
+	if err != nil {
+		t.Errorf("Could not get primary image handle: %s", err)
+	}
+	if !handle.IsPrimaryImage() {
+		t.Error("Expected primary image")
+	}
+
+	exifCount := handle.ExifCount()
+	fmt.Println("Exif count", exifCount)
+
+	exifIDs := handle.ExifIDs()
+	fmt.Println("Exif IDs", exifIDs)
+
+	metadataCount := handle.MetadataCount()
+	fmt.Println("Metadata count", metadataCount)
+
+	metadataIDs := handle.MetadataIDs()
+	fmt.Println("Metadata IDs", metadataIDs)
+
+	metadataID := metadataIDs[0]
+
+	times, err := handle.ImageTimes(metadataID)
+	if err != nil {
+		t.Fail()
+	}
+	fmt.Println(times)
+
+	thumbnail := false
```

Comment:
The rest of the test, from here on and out, is the same as in heif_test.go, and can potentially be skipped.

Reply:
I added code that skips this, and also skips the entire test if the example image (which may be copyrighted) does not exist in the repository.